### ISSUE TITLE
Multi-GPU training with accelerate，but on one gpu

### ISSUE THREAD

**Issue by robot-feng** [huggingface/lerobot#2407]
### System Info

```Shell
ubuntu20,lerobot4.1
```

### Information

- [ ] One of the scripts in the examples/ folder of LeRobot
- [ ] My own task or dataset (give details below)

### Reproduction

#!/bin/bash

# Multi-GPU training with accelerate
# Using GPUs: 0, 1, 3 (3 processes total)

# Activate conda environment
source ~/anaconda3/etc/profile.d/conda.sh
conda activate lerobot
export MUJOCO_GL=egl
export PYOPENGL_PLATFORM=egl
export TOKENIZERS_PARALLELISM=false
export HF_HUB_DOWNLOAD_TIMEOUT=120
export HF_HUB_ETAG_TIMEOUT=120
export HF_ENDPOINT=https://huggingface.co  # 使用官方源避免镜像站问题
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True  # 减少内存碎片
export CUDA_VISIBLE_DEVICES=1,2,3
export NCCL_P2P_DISABLE=1
export NCCL_IB_DISABLE=1
export TOKENIZERS_PARALLELISM=false

# ================================================================================================
# 多GPU训练 - 从预训练SmolVLA模型开始(推荐用于MetaWorld)
#  cat src/lerobot/envs/metaworld_config.json | grep -A 30 "DIFFICULTY_TO_TASKS" 查看任务列表

accelerate launch \
--num_machines=1 \
--num_processes=3 \
src/lerobot/scripts/lerobot_train.py \
--policy.type=smolvla \
--policy.load_vlm_weights=true \
--policy.repo_id=HuggingFaceTB/SmolVLM-500M-Instruct \
--policy.max_state_dim=256 \
--policy.num_steps=1 \
--dataset.repo_id=HuggingFaceVLA/libero \
--batch_size=4 \
--num_workers=4 \
--steps=2000000 \
--log_freq=100 \
--save_freq=2000 \
--env.type=libero \
--env.task=libero_10 \
--eval_freq=2000 \
--eval.batch_size=1 \
--eval.n_episodes=1 \
--output_dir=outputs/train/smolvla_libero \
--job_name=smolvla_libero_training \
--wandb.enable=true \
--wandb.project=lerobot-libero \
--wandb.notes="ada loss" \
--seed=1000 > accelerate_run.log 2>&1 &

<img width="1016" height="416" alt="Image" src="https://github.com/user-attachments/assets/5f0b3f78-bf20-493b-a7cb-e41388f8660a" />

### Expected behavior

should every gpu equal

**Comment by robot-feng** [comment]
<img width="1530" height="234" alt="Image" src="https://github.com/user-attachments/assets/2069cd39-16c8-4701-b5c8-ba704979ab95" />

**Comment by icsl-Jeon** [comment]
Same here

**Comment by WenOOI** [comment]
Same here

**Comment by WenOOI** [comment]
I found that using accelerate for multi-gpu training with evaluation arguments causes this issue. Removing the following arguments in the training script resolved it for me:

--env.type=libero
--env.task=libero_10
--eval_freq=2000
--eval.batch_size=1
--eval.n_episodes=1


update: After switching to a different machine and using the latest repository code, I haven't encountered this problem again.

**Comment by imstevenpmwork** [comment]
Hello! Can you guys confirm this is still an issue in latest `main`?
